From 49a98eaa2d20bb3498dc390d4cf4b051cff633fa Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Mon, 23 Apr 2018 16:06:41 +0300
Subject: [PATCH 3620/5242] MLK-18115: drm: imx: dcss: fix usage of
 uninitialized variable

commit  b70e11b7059f7a74721df7a19ef6015211113162 from
https://source.codeaurora.org/external/imx/linux-imx.git

In certain conditions, i.e. YUV mode, pixel_depth variable will be used
uninitialized. This can lead to unpredictable behavior.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-scaler.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-scaler.c b/drivers/gpu/imx/dcss/dcss-scaler.c
index db5638c..48242a5 100644
--- a/drivers/gpu/imx/dcss/dcss-scaler.c
+++ b/drivers/gpu/imx/dcss/dcss-scaler.c
@@ -914,7 +914,7 @@ void dcss_scaler_setup(struct dcss_soc *dcss, int ch_num, u32 pix_format,
 	enum dcss_color_space dcss_cs;
 	int planes;
 	const struct drm_format_info *format;
-	unsigned int pixel_depth;
+	unsigned int pixel_depth = 0;
 	bool rtr_8line_en = false;
 	bool use_5_taps = false;
 	enum buffer_format src_format = BUF_FMT_ARGB8888_YUV444;
-- 
1.7.9.5

